src/com/projectkorra/projectkorra/command/DisplayCommand.java

```diff
@@ -349,63 +318,106 @@ private void displayBinds(final CommandSender sender) {
+			if (!coreAbil.getElement().equals(element) || !coreAbil.isEnabled()|| coreAbil.isHiddenAbility() || abilities.contains(coreAbil.getName()) || coreAbil instanceof ComboAbility || coreAbil instanceof PassiveAbility) {
+				continue;
+			}
+			abilities.add(coreAbil.getName());
+		}
+		return abilities;
+	}
+	
+	private List<String> getCombos(final Element element) {
+		if (element instanceof SubElement) {
+			final List<String> combos = new ArrayList<>();
+			for (CoreAbility coreAbil : CoreAbility.getAbilitiesByElement(element)) {
+				if (!(coreAbil instanceof ComboAbility) || !coreAbil.isEnabled() || coreAbil.isHiddenAbility() || combos.contains(coreAbil.getName())) {
+					continue;
 				}
-
-				sender.sendMessage(message);
+				combos.add(coreAbil.getName());
 			}
+			return combos;
+		} else {
+			return ComboManager.getCombosForElement(element);
 		}
 	}
 
 	@Override
 	protected List<String> getTabCompletion(final CommandSender sender, final List<String> args) {
-		if (args.size() >= 1 || !sender.hasPermission("bending.command.display")) {
-			return new ArrayList<String>();
+		if (!args.isEmpty() || !sender.hasPermission("bending.command.display")) {
+			return new ArrayList<>();
+		}
+		final List<String> list = new ArrayList<>();
+		
+		for (final Element e : Element.getMainElements()) {
+			list.add(e.getName());
+			for (final SubElement s : Element.getSubElements(e)) {
+				list.add(s.getName());
+			}
 		}
-		final List<String> list = new ArrayList<String>();
-		list.add("Air");
-		list.add("Earth");
-		list.add("Fire");
-		list.add("Water");
-		list.add("Chi");
 
 		for (final Element e : Element.getAddonElements()) {
 			list.add(e.getName());
+			for (final SubElement s : Element.getAddonSubElements(e)) {
+				list.add(s.getName());
+			}
 		}
 
-		list.add("Bloodbending");
-		list.add("Combustion");
-		list.add("Flight");
-		list.add("Healing");
-		list.add("Ice");
-		list.add("Lava");
-		list.add("Lightning");
-		list.add("Metal");
-		list.add("Plantbending");
-		list.add("Sand");
-		list.add("Spiritual");
-		list.add("BlueFire");
-
-		for (final SubElement se : Element.getAddonSubElements()) {
-			list.add(se.getName());
+		for (String elementName : new ArrayList<>(list)) {
+			list.add(elementName + "Combos");
+			list.add(elementName + "Passives");
 		}
 
-		list.add("AirCombos");
-		list.add("EarthCombos");
-		list.add("FireCombos");
-		list.add("WaterCombos");
-		list.add("ChiCombos");
-		list.add("Avatar");
-
-		list.add("AirPassives");
-		list.add("EarthPassives");
-		list.add("FirePassives");
-		list.add("WaterPassives");
-		list.add("ChiPassives");
-
 		return list;
 	}
-}
+	
+	private void fillAbbreviations() {
+		abbreviations.put("ac", "aircombo");
+		abbreviations.put("ap", "airpassive");
+		abbreviations.put("avc", "avatarcombo");
+		abbreviations.put("avp", "avatarpassive");
+		abbreviations.put("cc", "chicombo");
+		abbreviations.put("cp", "chipassive");
```

Comment:
I wonder if you should make a public method to add custom abbreviations, too. Just for things like custom elements

Reply:
Good point, will do that!

Reply:
Added see DisplayCommand#addAbbreviation()